extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderHooks.class.php

```diff
@@ -46,41 +46,31 @@ public static function onWikiaSkinTopScripts( &$vars, &$scripts, $skin ) {
 		$title = $skin->getTitle();
 
 		if ( $title && $title->isSpecial( PortableInfoboxBuilderSpecialController::PAGE_NAME ) ) {
-			$templateTitleText = self::getUrlPath( $title->getText() );
+			$templateTitleText = PortableInfoboxBuilderHelper::getUrlPath( $title->getText() );
 
 			// We need the variable only if Infobox Builder launches (there is a template title provided)
 			if ( $templateTitleText ) {
 				$templateTitle = Title::newFromText( $templateTitleText, NS_TEMPLATE );
 				$vars['templatePageUrl'] = $templateTitle->getFullUrl();
+				$vars['sourceEditorUrl'] = $templateTitle->getFullUrl( [
+					'action' => 'edit',
+					'useeditor' => 'source'
+				] );
 			}
 		}
 
 		return true;
 	}
 
 	/**
-	 * remove the special page name from the title and return name of the template
-	 * passed after the slash without the namespace, i.e.
-	 * Special:InfoboxBuilder/TemplateName/Subpage => TemplateName/Subpage
-	 * @param $titleText
-	 * @return string
-	 */
-	private static function getUrlPath( $titleText ) {
-		return implode( '/', array_slice( explode( '/', $titleText ), 1 ) );
-	}
-
-	/**
 	 * Add global variables for Javascript
 	 * @param array $aVars
 	 * @return bool
 	 */
 	public function onEditPageMakeGlobalVariablesScript( array &$aVars ) {
 		$context = \RequestContext::getMain();
 		$title = $context->getTitle();
-		if (
-			( new \Wikia\TemplateClassification\Permissions() )->shouldDisplayEntryPoint( $context->getUser(), $title )
-			&& \RequestContext::getMain()->getRequest()->getVal( 'action' ) === 'edit'
-		) {
+		if ( self::shouldPassInfoboxBuilderVars( $context ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)